modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/io/AbstractDataPageIO.java

```diff
@@ -386,16 +391,18 @@ public int getRowsCount(long pageAddr) {
     /**
      * @param pageAddr Page address.
      * @param c Closure.
-     * @param <T> Closure return type.
+     * @param <U> Closure return type.
      * @return Collection of closure results for all items in page.
      * @throws IgniteCheckedException In case of error in closure body.
      */
-    public <T> List<T> forAllItems(long pageAddr, CC<T> c) throws IgniteCheckedException {
+    public <U> List<U> forAllItems(long pageAddr, CC<U> c) throws IgniteCheckedException {
```

Comment:
I don't see the need to change T to U.

Reply:
<T> on the method shadows with <T> on the class. It brings some confusion, so it seems that it's better to name them differently.

Reply:
Let's not change the current code.

Reply:
This is an improvement. It removes some confusion that was in the code. It does not add any problems (at least I don't see them, and the compiler is happy).

Why would you like to retain the confusion?